spec/features/users/user_can_delete_an_answer_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+feature 'user can delete an answer' do
+  scenario 'user owns an answer and clicks "Delete Answer"' do
+    user = Fabricate(:user, id: 1, name: "we_the_best")
+    user.roles.create(name: 'registered_user')
+
+    question = Fabricate(:question,
+                                      title: "Why are we the best?",
+                                      body: "They're trying to stop us.",
+                                      user: user)
+
+    answer = question.answers.create(body: "BlessUp", user: user)
+    answer_comment = answer.comments.create(body: "DJKhaled", user: user)
+    question_comment = question.comments.create(body: "Gratitude", user: user)
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    visit question_path(question)
+
+    within(".question-answers .question-answer:nth-child(1)") do
+      expect(page).to have_content("BlessUp")
+    end
+
+    first('#answer-delete-button').click_button('Delete Answer')
+
+    expect(current_path).to eq(question_path(question))
+
```

Comment:
Awesome assertions on this. It makes the test really show what exactly happened when the button was clicked.